[Fan Tianhao] iP

TTraveller7 opened the pull request:
### Duke

> "Your mind is for having ideas, not holding them." - David Allen ([Source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😄 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

TTraveller7 committed:
Level-1 implementation

TTraveller7 committed:
level-2 implementation

TTraveller7 committed:
level-3 implementation

TTraveller7 committed:
level-4 implementation

TTraveller7 committed:
level-4 implementation

TTraveller7 committed:
level-5 implementation

TTraveller7 committed:
Merge branch 'master' of https://github.com/TTraveller7/ip

TTraveller7 committed:
level-6 implementation

TTraveller7 committed:
level-7 Implementation

TTraveller7 committed:
level-8 implementation

TTraveller7 committed:
Merge branch 'branch-level-8'

TTraveller7 committed:
A-TestUiTesting partial Implementation

luayida99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,321 @@
+
+    // UI
+
+    private static void PrintWithIndent(String s) {
```
Method name should be in camelCase, consider changing it to `printWithIndent` ?

luayida99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,321 @@
+        System.out.println(LIndent + s);
+    }
+
+    private static void PrintList() {
```
Same thing with camelCase method names, consider changing it to `printList` ?

luayida99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,321 @@
+        }
+    }
+
+    private static boolean IsMarkDown(String s) throws DukeException {
```
```suggestion
    private static boolean isMarkDown(String s) throws DukeException {
```

luayida99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,321 @@
+        return false;
+    }
+
+    private static boolean IsDelete(String s) throws DukeException {
```
```suggestion
    private static boolean isDelete(String s) throws DukeException {
```

luayida99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,321 @@
+        return false;
+    }
+
+    private static String[] ParseDeadlineString(String s) throws DukeException {
```
```suggestion
    private static String[] parseDeadlineString(String s) throws DukeException {
```

luayida99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,321 @@
+                : new String[] {s.substring(0, index), s.substring(index + DeadlineWord_By.length())};
+    }
+
+    private static String[] ParseEventString(String s) throws DukeException {
```
```suggestion
    private static String[] parseEventString(String s) throws DukeException {
```

luayida99 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    @Override
+    public String PopulateSaveData() {
```
```suggestion
    public String populateSaveData() {
```

luayida99 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+        return "[" + this.getStatusIcon() + "] " + description;
+    }
+
+    public String PopulateSaveData() {
```
```suggestion
    public String populateSaveData() {
```

luayida99 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+        return "T | " + (isDone ? 1 : 0) + " | " + description;
+    }
+
+    public static Task LoadFromSaveData(String s) throws DukeException{
```
```suggestion
    public static Task loadFromSaveData(String s) throws DukeException {
```

luayida99 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+
+    public static Task LoadFromSaveData(String s) throws DukeException{
+        String[] arr = s.split(dividerWord, 4);
+        switch (arr[0]) {
+            case "T":
+                ToDo todo = new ToDo(arr[2]);
+                todo.isDone = arr[1].equals("1");
+                return todo;
+            case "D":
+                Deadline deadline = new Deadline(arr[2], arr[3]);
+                deadline.isDone = arr[1].equals("1");
+                return deadline;
+            case "E":
+                Event event = new Event(arr[2], arr[3]);
+                event.isDone = arr[1].equals("1");
+                return event;
+            default:
+                throw new DukeException("Unknown symbol in save file.");
```
Could remove the indentation in front of the case statements? Coding standard states that switch statements shouldn't have indentation in front of case statements!

luayida99 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,19 @@
+    }
+
+    @Override
+    public String PopulateSaveData() {
```
```suggestion
    public String populateSaveData() {
```

luayida99 reviewed (commented):
Overall, code was well written and easy to understand! Some nits found, please take a look at my comments and make any changes you deem necessary.

TTraveller7 committed:
Implement A-MoreOOP level

TTraveller7 committed:
Implement A-Packages level

TTraveller7 committed:
Reverse to a former branch

TTraveller7 committed:
Merge branch 'master' of https://github.com/TTraveller7/ip

TTraveller7 committed:
Implement A-JUnit level

TTraveller7 committed:
Implement A-Jar

TTraveller7 committed:
Adds java doc.

TTraveller7 committed:
Check code with coding standard.

TTraveller7 committed:
Implements level-9

TTraveller7 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/duke/DukeException.java
#	src/main/duke/Ui.java

TTraveller7 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/duke/Parser.java

TTraveller7 committed:
Add comments.

TTraveller7 committed:
Merge branch 'add-gradle-support'

hmanhduc2k commented on src/main/java/Duke.java:
```diff
@@ -108,6 +116,91 @@ public static void main(String[] args) {
+            PrintWithIndent("Error occurs when : " + e);
+        }
+    }
+
```
This two methods could be written in a separate java file instead for more OOP, and both of them can also be simplified for ease of reading (in terms of implementation), and to reduce your class implementation length.

TTraveller7 committed:
Change directories.

hmanhduc2k commented on src/main/java/Duke.java:
```diff
@@ -108,6 +116,91 @@ public static void main(String[] args) {
+
+    private static void InitializeSaveData() {
+        try {
+            Path path = Arrays.stream(SaveDirectory)
```
Change your method declaration to initializeSaveData. Try to follow the standard set by CS2103T in naming your functions

hmanhduc2k commented on src/main/java/Task.java:
```diff
@@ -15,4 +16,28 @@ public String getStatusIcon() {
+                Event event = new Event(arr[2], arr[3]);
+                event.isDone = arr[1].equals("1");
+                return event;
+            default:
```
Try to follow CS2103T guideline on coding standard - in a sense that switch and case should be of the same indentation level.

hmanhduc2k commented on src/main/java/Deadline.java:
```diff
@@ -10,4 +10,9 @@ public Deadline(String description, String by) {
+    @Override
+    public String PopulateSaveData() {
+        return "D | " + (this.isDone ? 1 : 0) + " | " + this.description + " | " + this.by;
+    }
```
Change your method declaration to match the naming standard, which means that this should be changed to populateSaveData() instead.

hmanhduc2k commented on src/main/java/ToDo.java:
```diff
@@ -3,8 +3,17 @@ public ToDo(String description) {
 
+    protected static ToDo stringToToDo(String s) {
+        return new ToDo(s.substring(12));
+    }
```
Try to add in JavaDoc in all of your methods. In particular, you probably will need JavaDoc for this method as it is difficult to understand what it is doing from the naming (to-to-do).

hmanhduc2k reviewed (commented):
Overall, great effort in following the module coding standard. However, I believe that it would be much better if you can add Java Documentations to all methods, change your naming standard, change how you structure your code, and possibly reduce your code size to make it readable.

TTraveller7 committed:
Merge branch 'add-gradle-support' into A-Gradle

TTraveller7 committed:
Add gradle to project

TTraveller7 committed:
Merge tag 'A-Gradle'

TTraveller7 committed:
Add package Command, which encapsulates user commands.

TTraveller7 committed:
Add JUnit tests on command classes.

TTraveller7 committed:
level-10 implementation

TTraveller7 committed:
Change the code to follow java coding standard.

Previous code has several places violating the coding standard. Some
public method lacks Javadoc.

Let's,
* avoid coding style violations
* add javadoc for non-trival public methods and constructors

TTraveller7 committed:
Add assertions and parameter checking.

Public methods need to throw exceptions when parameters are invalid.
Nonpublic methods need to check program state at runtime.

Let's,
* add parameter checking for most of the public methods and
  constructors
* add assertions to identify bugs

TTraveller7 committed:
Add the ability to recognize and deal with duplicate task.

User is not allowed to create a new task if it equals to an existing
unfinished task.

Other improvements:
* Reduce "arrow-head" methods.
* Add welcome message in GUI
* Change exception type.

TTraveller7 committed:
Minor improvements on variable naming and javadoc format.

TTraveller7 committed:
Improve code quality.

Let's:
* add javadoc for public methods;
* avoid repeat code;
* avoid deep nesting;
* remove unused methods.

TTraveller7 committed:
Merge pull request #2 from TTraveller7/branch-A-CodeQuality

Improve code quality.

TTraveller7 committed:
Improve GUI.

Modifications:
* add background image;
* change fonts;
* make window auto resize;
* change user image;
* change UI characteristic.

TTraveller7 committed:
* Refactor class name from duke to pats;
* Add user guide;
* Add Ui image.

TTraveller7 committed:
Update README.md

TTraveller7 committed:
Update README.md

TTraveller7 committed:
Minor fix of file name.

TTraveller7 committed:
no message

TTraveller7 committed:
Minor fix on file name.

TTraveller7 committed:
Add linux & mac dependencies.
